File: java/08_pgm_invoice_engine/src/co/edu/campusucc/poo/domain/Invoice.java
package co.edu.campusucc.poo.domain;

import co.edu.campusucc.poo.engines.InvoiceEngine;
import co.edu.campusucc.poo.interfaces.Iinvoice;

public class Invoice extends InvoiceEngine implements Iinvoice {

    @Override
    public void tipo() {
        System.out.println("Tipo: Factura Fisica 📃📃");
    }

    @Override
    public void total() {
        System.out.println("Total:💲200 ");
    }

    @Override
    public void id() {
        System.out.println("🆔: 3672");
    }
    
}

File: c++/08_pgm_invoice_engine/src/co/edu/campusucc/poo/domain/Invoice.cpp
#include <iostream>
#include "InvoiceEngine.h"
#include "Iinvoice.h"

using namespace std;

class Invoice : public InvoiceEngine, public Iinvoice {
public:
    void tipo() {
        cout << "Tipo: Factura Fisica 📃📃" << endl;
    }

    void total() {
        cout << "Total:💲200 " << endl;
    }

    void id() {
        cout << "🆔: 3672" << endl;
    }
};
